module/dice/damage-roll.mjs

```diff
@@ -302,7 +302,9 @@ export default class DamageRoll extends BasicRoll {
     );
     const DialogClass = this.DefaultConfigurationDialog;
     return await DialogClass.configure(
-      { critical: { allow: allowCritical } }, { options: { title } }, { rollMode: defaultRollMode }
+      { critical: { allow: allowCritical }, rolls: rolls.map(r => ({ parts: [r.formula], options: r.options })) },
```

Comment:
This is fixing a bug in `4.0` where we didn't pass in any information on the actual rolls to the dialog, so nothing would actually be rolled. We need to split into `parts` and `options` because the dialog is expecting `BasicRollProcessConfiguration`, which normally contains the rolls before they are fully created. In this case I am backwards engineering that config object from rolls themselves.

Reply:
I somehow completely glossed over that this was a deprecated method. It makes sense we would have to this sort of backwards engineering in order to support the deprecated signature.

Unrelated, but I didn't want to miss it: It seems `RollConfigurationDialog.configure` is annotated as accepting a `BasicRollConfigurationDialogOptions` as its second parameter (`dialog`), but then passes `dialog.options` to the constructor. Is it actually expecting to receive a `BasicRollDialogConfiguration` as its second parameter instead?

Reply:
You are correct.